compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1016,13 +1015,31 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         (result, dep_node)
     }
 
-    // Treat negative impls as unimplemented, and reservation impls as ambiguity.
-    fn filter_negative_and_reservation_impls(
+    fn filter_impls(
         &mut self,
         candidate: SelectionCandidate<'tcx>,
+        obligation: &TraitObligation<'tcx>,
     ) -> SelectionResult<'tcx, SelectionCandidate<'tcx>> {
+        let tcx = self.tcx();
+        // Respect const trait obligations
```

Comment:
does `self` give us enough information to check if we are in a const context?

Reply:
If not, we could start tracking it.

Reply:
Done